icu4j/main/core/src/main/java/com/ibm/icu/text/RuleBasedNumberFormat.java

```diff
@@ -1350,13 +1319,12 @@ public Number parse(String text, ParsePosition parsePosition) {
             workingPos.setIndex(0);
         }
 
-        // add the high water mark to our original parse position and
+        // add the high watermark to our original parse position and
```

Comment:
I think this should actually be "high-water mark"...

Reply:
The code editor that I was using does spelling and grammar checking. It highlighted this along with the other oddities, like inro, and ned.

It seems like a valid English word to me. https://en.wiktionary.org/wiki/watermark

Reply:
Actually the original was right. I'll revert this change.

https://en.wikipedia.org/wiki/High_water_mark